src/richie/apps/courses/admin.py

```diff
@@ -44,18 +44,6 @@ def title(self, obj):
         """
         return obj.extended_object.get_title()
 
-    def save_related(self, request, form, formsets, change):
-        """
-        When a model is saved via the admin, the main object gets saved first (to make sure it has
-        a pk), then the many-to-many are cleared and the new values set to whatever came out of
-        the form (see http://stackoverflow.com/a/1925784/469575).
-
-        This means that we can not rely on our override of the `save` method on the model to modify
-        the `organizations` many-to-many field and make sure it includes the main organization.
-        """
-        super().save_related(request, form, formsets, change)
-        form.instance.organizations.add(form.instance.organization_main)
```

Comment:
Removing this make it so that choosing a organization_main for a new course does not also add this main org to attached organizations.

Reply:
Yeah I removed it at some point but we should be able to keep this feature with the final solution I found. Let me revert this together with @mehdi suggestion so you can it.